src/PaymentGateways/Gateways/Stripe/CheckoutGateway.php

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * @unreleased
+ */
+class CheckoutGateway extends PaymentGateway
+{
+    use Traits\CheckoutInstructions;
+    use Traits\CheckoutModal;
+    use Traits\HandlePaymentIntentStatus;
+
+    /**
+     * @inheritDoc
+     * @unreleased
+     * @return GatewayCommand
+     * @throws PaymentGatewayException
+     */
+    public function createPayment( GatewayPaymentData $paymentData )
+    {
+        $workflow = new Workflow();
+        $workflow->bind( $paymentData );
+
+        $workflow->action( new Actions\SaveDonationSummary );
+        $workflow->action( new Actions\GetOrCreateStripeCustomer );
+
+        switch (give_stripe_get_checkout_type()) {
+            case 'modal':
+                return $this->createPaymentModal($workflow);
+            case 'redirect':
+                return $this->createPaymentRedirect($workflow);
+            default:
+                throw new CheckoutException('Invalid Checkout Error');
+        }
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param $workflow
+     * @return PaymentProcessing|RedirectOffsite
+     * @throws Exceptions\PaymentIntentException
+     */
+    protected function createPaymentModal($workflow)
+    {
+        $workflow->action(new Actions\GetPaymentMethodFromRequest);
+        $workflow->action(new Actions\CreatePaymentIntent());
+        $paymentIntent = $workflow->resolve(PaymentIntent::class);
+        return $this->handlePaymentIntentStatus($paymentIntent);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param $workflow
+     * @return RedirectOffsite
+     */
+    protected function createPaymentRedirect($workflow)
+    {
+        $workflow->action(new Actions\CreateCheckoutSession());
+        $session = $workflow->resolve(CheckoutSession::class);
+        return new RedirectOffsite(
+            CheckoutHelper::getRedirectUrl($session->id(), give_get_payment_form_id($paymentData->donationId))
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function id()
+    {
+        return 'stripe_checkout';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getId()
+    {
+        return self::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return __('Stripe - Checkout', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPaymentMethodLabel()
+    {
+        return __('Stripe - Checkout', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getLegacyFormFieldMarkup($formId, $args)
+    {
+        Stripe::canShowBillingAddress( $formId, $args );
+
+        if (FormUtils::isLegacyForm($formId)) {
+            return false;
+        }
+
+        switch( give_stripe_get_checkout_type() ) {
+            case 'modal':
+                return $this->getCheckoutModalHTML( $formId, $args );
+            case 'redirect':
+                return $this->getCheckoutInstructions();
+        }
+    }
```

Comment:
@kjohnson i'm not seeing any instructions here, am I missing something?

![Screen Shot 2022-02-04 at 9 42 14 AM](https://user-images.githubusercontent.com/10138447/152554630-5ff6371c-1dc8-49ea-8cad-f2ee60d79f53.png)

Reply:
Hm... it seems that `develop` uses the same redirect notice regardless of the modal/redirect setting. So the HTML rendered in `develop` when set to `modal` (default) is not correct.

Reply:
For now, I'll restore the current expectation and we can refactor from there.

Reply:
@jonwaldstein resolved, see https://github.com/impress-org/givewp/pull/6199/commits/a0710523d8095cba857d51f9c5193871179f0875